chainstate/src/detail/mod.rs

```diff
@@ -744,28 +748,38 @@ impl<S: BlockchainStorage, V: TransactionVerificationStrategy> Chainstate<S, V>
         &self.events_controller
     }
 
-    pub fn is_initial_block_download(&self) -> Result<bool, PropertyQueryError> {
-        if self.is_initial_block_download_finished {
-            return Ok(false);
+    pub fn is_initial_block_download(&self) -> bool {
+        !self.is_initial_block_download_finished.test()
+    }
+
+    /// Returns true if the given block timestamp is newer than `ChainstateConfig::max_tip_age`.
+    fn is_fresh_block(&self, time: &BlockTimestamp) -> bool {
+        let now = self.time_getter.get_time();
+        time.as_duration_since_epoch() + self.chainstate_config.max_tip_age.clone().into() > now
+    }
+
+    /// Update `is_initial_block_download_finished` when tip has changes (can only be set once)
+    fn update_initial_block_download_flag(&mut self) -> Result<(), PropertyQueryError> {
```

Comment:
"when tip has changes" sounds broken.
Perhaps, it should be something like "if we've got a fresh tip"?

Reply:
I think it's typo from "when tip has changed", or "when tip changes"

Reply:
Corrected, thanks.